### ISSUE TITLE
[Bug][AMD][Upstream] Bus Error crash when teleporting in The Legend of Zelda: Tears of the Kingdom

### ISSUE THREAD

**Issue by GraithTiger** [Ryujinx/Ryujinx#6822]
### Description of the issue

Launching and initial save load works fine, but teleporting to a different error causes a freeze and eventual crash once load is finished.

Process exits with `Bus error (core dumped)`

Crash occurs whether memory management is set to Software, Host, or Host unsafe

### Reproduction steps

From the depths, teleport to a shrine or tower in the overworld.

### Log file

[Ryujinx_1.1.1312_2024-05-17_00-23-24.log](https://github.com/Ryujinx/Ryujinx/files/15344115/Ryujinx_1.1.1312_2024-05-17_00-23-24.log) - Optimizer mods, trace logging disabled
[Ryujinx_1.1.1312_2024-05-17_00-33-36.log.tar.gz](https://github.com/Ryujinx/Ryujinx/files/15344128/Ryujinx_1.1.1312_2024-05-17_00-33-36.log.tar.gz) - No mods, default settings with trace logging enabled

### OS

Arch Linux 6.8.9-arch1-2

### Ryujinx version

1.1.1312

### Game version

1.2.1

### CPU

AMD Ryzen 7 3900X

### GPU

AMD Radeon RX 6800XT

### RAM

32GB

### List of applied mods

TOTK Optimizer 60FPS / none

**Comment by steven-a-wilson** [comment]
I'm able to consistently reproduce this on a similar configuration (Arch Linux, kernel 6.8.9-zen1-2-zen, Ryzen 9 3900X, RX 5700 XT). Interestingly, I got bus error crashes with both RADV and AMDVLK. The time taken between the teleport and the crash seems to vary somewhat; in one case it happened even before the loading screen completely disappeared, and in others it took a minute or so of running around Lookout Landing afterward.

I did not get the crash on the other machine I have handy at the moment (Fedora 40, kernel 6.8.8-300.fc40.x86_64, Ryzen 5 5600X, GTX 970).

(I'm partly interested in this because I've been sporadically getting bus error crashes and ring gfx timeouts with Breath of the Wild, which I'm guessing might have the same root cause, but I didn't find a consistent way to reproduce it.)
